lib/v2/bilibili/utils.js

```diff
@@ -65,20 +66,35 @@ function hexsign(e) {
     return o;
 }
 
-function addVerifyInfo(params, verifyString) {
+function addWbiVerifyInfo(params, wbiVerifyString) {
     const searchParams = new URLSearchParams(params);
     searchParams.sort();
     const verifyParam = searchParams.toString();
     const wts = Math.round(Date.now() / 1000);
-    const w_rid = md5(`${verifyParam}&wts=${wts}${verifyString}`);
+    const w_rid = md5(`${verifyParam}&wts=${wts}${wbiVerifyString}`);
     return `${params}&w_rid=${w_rid}&wts=${wts}`;
 }
 
+function getDmImgList() {
+    if (typeof (config.bilibili.dmImgList) === 'undefined') {
+        return '[{"x":2721,"y":615,"z":0,"timestamp":29,"type":0}]';    // This is true data from browser
```

Comment:
They are simply the cursor locations of which these events happened.
<details><summary>Events</summary>

```js
var p = ["click", "mousemove", "keydown", "DOMMouseScroll", "mousewheel", "mousedown", "touchstart", "touchmove", "focus"]
```
</details>

<details><summary>log stack</summary>

```js
F = function(t) {
        var e = N
          , n = function(t, e, n, r, o, i) {
            return t(e, n, r, o, i)
        }
          , r = t[e(519)]((function(t, e) {
            var r = t.x
              , o = void 0 === r ? 0 : r
              , i = t.y
              , a = void 0 === i ? 0 : i
              , c = t.timestamp
              , u = t.type;
            return n(B, o, a, c, u, e)
        }
        ))[e(519)]((function(t) {
            return {
                x: t[0],
                y: t[1],
                z: t[2],
                timestamp: t[3],
                type: t[4]
            }
        }
        ));
        try {
            return JSON.stringify(r)
        } catch (t) {
            return console[e(541)](t),
            ""
        }
    }
```
</details>

I would prefer generating a random one instead.

Reply:
Sorry, I really don't know how to generate it correctly.

I have tried to generate it from Gaussian Distributed, but it didn't work.

```js
// https://github.com/errcw/gaussian/blob/master/lib/box-muller.js
function generateGaussianInteger(mean, std) {
    const _2PI = Math.PI * 2;
    const u1 = Math.random();
    const u2 = Math.random();

    const z0 = Math.sqrt(-2.0 * Math.log(u1)) * Math.cos(_2PI * u2);

    return Math.round(z0 * std + mean);
}

function getDmImgList() {
    const path = [
        {
            x: Math.max(generateGaussianInteger(2800, 100), 0),
            y: Math.max(generateGaussianInteger(1000, 200), 0),
            z: 0,
            timestamp: Math.max(generateGaussianInteger(50, 5), 0),
            type: 0,
        },
    ];
    return JSON.stringify(path);
    // if (typeof (config.bilibili.dmImgList) === 'undefined') {
    //     return '[{"x":2721,"y":615,"z":0,"timestamp":29,"type":0}]';    // This is true data from browser
    // }
    // return config.bilibili.dmImgList;
}
```

Reply:
I think the hints are quite obvious in https://github.com/DIYgod/RSSHub/pull/14128#discussion_r1437770620.
<details><summary>Anyways, these are the related codes dumped from 9.space.*.js</summary>

```js
function G() {
    var t = ["szSez", "103591TJTTxc", "CjXyY", "CRCsi", "handler", "length", "map", "TXMjE", "189vkvYeF", "AWumG", "HwlZR", "init", "324378lGstCz", "valueMap", "WKvHl", "name", "3716570VOnHWp", "TBybc", "wcSXW", "mhLTH", "webglStr", "14nRDWOh", "error", "AioYO", "buffer", "2550606OYIlVp", "keys", "XGGLO", "log", "params", "KfRTB", "AHjCN", "92152cwuYJw", "rAndRender", "bdUwC", "webglVendo", "18zWDGOL", "tkpid", "stringify", "encode", "LOG", "payload", "forEach", "apply", "eFIXr", "34444817xwEJrQ", "ytHiH", "request", "hQAjN", "LsOHu", "substring", "userLogStr", "CdcgQ", "PsWqo", "MmJfn", "OdIfM", "ImqLE", "DhwGv", "4274880qQISqI", "qmlRV", "LztiW", "jmfws", "queryUserL", "ZHgrM", "4KGZfal"];
    return (G = function() {
        return t
    }
    )()
}

var N = M;
function M(t, e) {
    var n = G();
    return (M = function (t, e) {
        return n[(t -= 498)];
    })(t, e);
}

var D = {
    mousemove: 0,
    click: 1,
    keydown: 2,
    wheel: 3,
    touch: 4,
    focus: 5,
};

function F(t) {
    var e = N,
        n = function (t, e, n, r, o, i) {
            return t(e, n, r, o, i);
        },
        r = t
            .map(function (t, e) {
                var r = t.x,
                    o = void 0 === r ? 0 : r,
                    i = t.y,
                    a = void 0 === i ? 0 : i,
                    c = t.timestamp,
                    u = t.type;
                return n(B, o, a, c, u, e);
            })
            .map(function (t) {
                return { x: t[0], y: t[1], z: t[2], timestamp: t[3], type: t[4] };
            });
    try {
        return JSON.stringify(r);
    } catch (t) {
        return console.log(t), '';
    }
}

function B(t, e, n, r, o) {
    var i = N,
        a = {
            CRCsi: function (t, e) {
                return t(e);
            },
            add: function (t, e) {
                return t + e;
            },
            add_1: function (t, e) {
                return t + e;
            },
            multiply: function (t, e) {
                return t * e;
            },
            minus: function (t, e) {
                return t - e;
            },
            CjXyY: function (t, e) {
                return t(e);
            },
            OdIfM: function (t, e) {
                return t(e);
            },
        },
        c = a.CRCsi(H, o);
    return [
        a.add(a.add_1(a.CRCsi(V, 1) * t, a.multiply(a.CRCsi(V, 0), e)), c),
        a.add(a.minus(a.CjXyY(V, 2) * t, a.multiply(a.OdIfM(V, 3), e)), c),
        c,
        n,
        D[r],
    ];
}

var W = ['Mg', 'Mw', 'NA', 'NQ']; // 2, 3, 4, 5
function V(t) {
    var e = N,
        n = {
            add: function (t, e) {
                return t + e;
            },
            bdUwC: function (t, e) {
                return t(e);
            },
        },
        r = n.add(W[t], '==');
    return n.bdUwC(Number, Buffer.from(r, 'base64').toString('ascii'));
}

function H(t) {
    var e = N,
        n = {
            multiply: function (t, e) {
                return t * e;
            },
        };
    return Math.floor(n.multiply(n.multiply(Math.random(), 114), t));
}

console.log(
    F([
        {
            type: 'mousemove',
            x: 0,
            y: 0,
            preX: 0,
            preY: 0,
            changeDistance: 0,
            timestamp: 0,
        },
    ]),
);
```
</details>

---

<details><summary>Simplified version</summary>

```js
const D = {
    mousemove: 0,
    click: 1,
    keydown: 2,
    wheel: 3,
    touch: 4,
    focus: 5,
};

const F = (t) => {
    var n = (func, x, y, timestamp, type, index) => {
        return func(x, y, timestamp, type, index);
    };
    var r = t
        .map((e, i) => {
            const x = e.x || 0;
            const y = e.y || 0;
            const timestamp = e.timestamp;
            const type = e.type;
            return n(B, x, y, timestamp, type, i);
        })
        .map((t) => {
            return { x: t[0], y: t[1], z: t[2], timestamp: t[3], type: t[4] };
        });
    try {
        return JSON.stringify(r);
    } catch (t) {
        return console.log(t), '';
    }
};

const B = (x, y, timestamp, type, index) => {
    const c = H(index);
    return [V[1] * x + V[0] * y + c, V[2] * x - V[3] * y + c, c, timestamp, D[type]];
};

// var W = ['Mg', 'Mw', 'NA', 'NQ'];
const V = [2, 3, 4, 5];

const H = (index) => {
    return Math.floor(Math.random() * 114 * index);
};

console.log(
    F([
        {
            type: 'mousemove',
            x: 0,
            y: 100,
            preX: 0,
            preY: 0,
            changeDistance: 0,
            timestamp: 0,
        },
    ]),
);
```
</details>